server/player/player.go

```diff
@@ -670,13 +641,25 @@ func (d *Data) GiveItem(newItem item.Data, db *sql.DB) error { // TODO: Refactor
 }
 
 // TakeItem from Data
-func (d *Data) TakeItem(id int32, slot int16, amount int16, invID byte) {
-
+func (d *Data) TakeItem(id int32, slot int16, amount int16, invID byte, db *sql.DB) item.Data {
 	item, err := d.getItem(invID, slot)
 	if err != nil {
-		return
+		log.Println(err)
```

Comment:
I think the `item.ID() == id` check should be here instead of channel_player.go. If either of these checks fail the error should be returned and handled by the calling function.